app/models/tax_form_due_date.rb

```diff
@@ -1,2 +1,14 @@
 class TaxFormDueDate < ApplicationRecord
+
+  validates_uniqueness_of   :name
+
+  def self.datasearch(search, entity_type, jurisdiction)
+    search = search.downcase if search.present?
+    due_date_results = where( "jurisdiction ILIKE ? OR entity_type ILIKE ? OR name ILIKE ?","%#{search}%", "%#{search}%", "%#{search}%")
```

Comment:
there's an error in your search logic. type in california 100

Reply:
you're not parsing the query paramters correctly.

Reply:
fixed the logic. created function parse_query_params. Now searching better than google!!!